library/std/src/thread/scoped.rs

```diff
@@ -0,0 +1,301 @@
+use super::{current, park, Builder, JoinInner, Result, Thread};
+use crate::fmt;
+use crate::io;
+use crate::marker::PhantomData;
+use crate::panic::{catch_unwind, resume_unwind, AssertUnwindSafe};
+use crate::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
+use crate::sync::Arc;
+
+/// A scope to spawn scoped threads in.
+///
+/// See [`scope`] for details.
+pub struct Scope<'env> {
+    data: ScopeData,
+    env: PhantomData<&'env mut &'env ()>,
```

Comment:
It might be worth adding a comment here to indicate what's going on with variance: I can imagine this being confusing, especially for those trying to learn from `std` to write their own abstractions.